ols/app/metrics/token_counter.py

```diff
@@ -54,43 +36,22 @@ def __init__(self, llm: LLM) -> None:
         self.token_counter.llm = llm  # actual LLM instance
         self.token_handler = TokenHandler()  # used for counting input and output tokens
 
-    def on_llm_start(
+    async def on_llm_new_token(
+        self,
+        token: str,
+        **kwargs: Any,
+    ) -> None:
+        """Compute token count when llm token is yielded."""
+        if token is not None and token != "":
+            self.token_counter.output_tokens += self.tokens_count(token)
+
+    async def on_llm_start(
         self, serialized: dict[str, Any], prompts: list[str], **kwargs: Any
     ) -> None:
         """Run when LLM starts running."""
         self.token_counter.llm_calls += 1
-        self.token_counter.input_tokens_counted = 0
         for p in prompts:
-            self.token_counter.input_tokens_counted += self.tokens_count(p)
-
-    def on_llm_end(self, response: LLMResult, **kwargs: Any) -> None:
-        """Run when LLM completes running."""
```

Comment:
This is not accessed in the current streaming at all -> hook on `on_llm_new_token`